7.0 pep8

laetitia-gangloff opened the pull request:
I try to set some module in pep8, the PR is not complete.

It gives an idea of the avancement. If you have any advice, do not hesitate !

Commit:
[pep8] - account_export_csv

Commit:
[pep8] - account_move_line_report_xls

Commit:
[pep8] - account_journal_report_xls

Commit:
[pep8] - account_financial_report_horizontal

Commit:
[pep8] - account_financial_report_webkit_xls

pedrobaeza commented on account_export_csv/__openerp__.py:
```diff
@@ -19,15 +20,16 @@
 
-    Add a wizard that allow you to export a csv file based on accounting journal entries 
+    Add a wizard that allow you to export a csv file based on accounting
+journal entries
```
Please keep same indentation

pedrobaeza commented on account_export_csv/__openerp__.py:
```diff
@@ -35,13 +37,14 @@
 
-    TODO: rearange wizard view with only one button to generate file plus define a selection list to select report type
+    TODO: rearange wizard view with only one button to generate file plus
+define a selection list to select report type
```
Please keep same indentation

pedrobaeza commented on account_export_csv/wizard/account_export_csv.py:
```diff
@@ -187,16 +207,20 @@ def _get_header_analytic(self, cr, uid, ids, context=None):
-                        sum(debit) as sum_debit,sum(credit) as sum_credit,sum(debit) - sum(credit) as balance
-                        from account_move_line 
+        cr.execute("""  select aac.code as analytic_code,
+        aac.name as analytic_name,
```
Please keep same indentation

pedrobaeza commented on account_financial_report_horizontal/report/__init__.py:
```diff
@@ -1,3 +1,3 @@
-#import account_balance
+# import account_balance
 import account_balance_sheet
```
You can put `from . import ...`

pedrobaeza commented on account_financial_report_horizontal/report/account_balance_sheet.py:
```diff
@@ -24,18 +24,19 @@
 
-import pooler
-from report import report_sxw
+import openerp.pooler as pooler
```
Do you need pooler? For what? Old method instead of `self.pool`?

pedrobaeza commented on account_financial_report_webkit_xls/__openerp__.py:
```diff
@@ -33,14 +33,14 @@
-    'update_xml' : [
-        'wizard/general_ledger_wizard_view.xml',   
-        'wizard/trial_balance_wizard_view.xml',   
+    'update_xml': [
```
Use `data` instead of `update_xml`

pedrobaeza commented on account_move_line_report_xls/report/move_line_list_xls.py:
```diff
@@ -276,12 +354,15 @@ def generate_xls_report(self, _p, _xs, data, objects, wb):
+              'account.move.line',
+              parser=move_line_xls_parser)
 
 # vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
You can remove this line. It's editor specific

pedrobaeza commented:
Hi, Laetitia, thanks for the great work. You can use autopep8 as a first step to reduce the number of PEP8 issues. I have put some comments inline for some conventions I have found on the diff review. They are not all, and thery are not imprescindible, but we can also change this.

Regards.

laetitia-gangloff commented:
Thank you for the review !
I will now continue the work.

Commit:
[pep8] - review on account_export_csv, account_move_line_report_xls, account_journal_report_xls, account_financial_report_horizontal, account_financial_report_webkit_xls

Commit:
[pep8] - account_financial_report

Commit:
[pep8] - account_financial_report_webkit

Commit:
[pep8] review on account_financial_report, account_financial_report_horizontal, account_financial_report_webkit_xls, account_journal_report_xls, account_move_line_report_xls

laetitia-gangloff commented:
The pep8 issue not resolved are in two case:
-F841 local variable 'XXXX' is assigned to but never used
-F821 undefined name 'XXXX'

For the first (F841) I am not enough confident to remove the concerned variable.
For the second (F821) name seems defined as global in other methods, I don't know best practice for this.

pedrobaeza commented:
Hi, Laetita, thanks for continuing with the work. For the two issues you mention, you can do:
- Remove variables safely for F841 issues, because flake8 detect correctly if the variables is not used, so there's no possibility of side effect.
- For this case, global it's a very dangerous option, but we need to modify full module if we want to change the approach, so what we can do to avoid F821 issue is to initialize these variables (ws, nbr_columns, etc), at the beginning of the file, outside any class, just after import sentences.

Please try and comment here any doubt.

Regards.

eLBati commented:
Hello @laetitia-gangloff and @pedrobaeza 

about F841 I see cases, like `debit_formula` at https://github.com/acsone/account-financial-reporting/blob/7.0-pep8/account_move_line_report_xls/report/move_line_list_xls.py#L353 , where flake8 is not able to understand that the variable is used by `col_specs_template`.
So, if I am correct, in this case flake8 is reporting a false negative and we probably need a way to handle it (by expliciting which errors should not be considered errors by travis_run_flake8?)

Other cases, like `exc` at https://github.com/acsone/account-financial-reporting/blob/7.0-pep8/account_financial_report_webkit/report/common_reports.py#L378 , you can remove them safely

pedrobaeza commented:
Well, I'm looking at the code, and the implementation is very opaque (short variables names, no comments nor docstrings...), but I don't see how that local variable is used in any of this lines:

```
        c_specs = map(lambda x: self.render(x, self.col_specs_template, 'totals'), wanted_list)
        row_data = self.xls_row_template(c_specs, [x[0] for x in c_specs])
        row_pos = self.xls_write_row(ws, row_pos, row_data, row_style=self.rt_cell_style_right)
```

BTW, what we need to do is to refactorize code if the case is that, because we can omit this error in the entire repo.

Regards.

eLBati commented:
I think it is used at [line 149](https://github.com/acsone/account-financial-reporting/blob/7.0-pep8/account_move_line_report_xls/report/move_line_list_xls.py#L149)

Commit:
[pep8] - correct F821 by defined global in self variable

Commit:
[pep8] - F841 - remove unused style, exc

laetitia-gangloff commented:
I fixed F821 by using self instead of global.
I fixed the easy F841 (style and exc).

I confirm debit_formula, credit_formula and bal_formula are used.

guewen commented:
> ./account_financial_report_webkit_xls/report/general_ledger_xls.py:234:21: F841 local variable 'bal_formula' is assigned to but never used
> ./account_financial_report_webkit_xls/report/trial_balance_xls.py:266:17: F841 local variable 'level_class' is assigned to but never used

Those are really not used

> ./account_move_line_report_xls/report/move_line_list_xls.py:353:9: F841 local variable 'debit_formula' is assigned to but never used
> ./account_move_line_report_xls/report/move_line_list_xls.py:356:9: F841 local variable 'credit_formula' is assigned to but never used
> ./account_move_line_report_xls/report/move_line_list_xls.py:359:9: F841 local variable 'bal_formula' is assigned to but never used
> ./account_journal_report_xls/report/nov_account_journal_xls.py:291:9: F841 local variable 'debit_formula' is assigned to but never used
> ./account_journal_report_xls/report/nov_account_journal_xls.py:294:9: F841 local variable 'credit_formula' is assigned to but never used
> ./account_journal_report_xls/report/nov_account_journal_xls.py:297:9: F841 local variable 'bal_formula' is assigned to but never used

They are used below with for example `self.render("debit_formula").
`self.render()` uses the locals of the previous frame that's why the linter can't see it is used.

The possibilities I see:
- Trick the linter to make it think it is used, I don't know if it is possible
- Refactor `report_xls`: maybe the best solution, but seems out of the scope for a "pep8-PR". Note that I opened #8 for replacing `xlwt` by `openpyxl`, so it could be done during this work
- Use a `# noqa` with a strong comment why it is there. This is not a solution to advocate for usually, but it would be pragmatic here: anyway the linter can't ensure the variable is used correctly because it is used as a string later. It would allow to merge this PR and continue with a more in-depth refactoring.

I would choose to add a `# noqa` until a better solution is found (refactoring).

Commit:
[pep8] - remove unused level_class and general_ledger_xls/bal_formula - use false unused in a logger call

laetitia-gangloff commented:
\o/ green !

So I remove variable that are really not used.
And for other I use the first possibility proposed by @guewen :
- Trick the linter to make it think it is used, I don't know if it is possible

It was the easier for me, I used the variable in logger. Variable contains string and logger was already declared. What do you think about this ?

pedrobaeza commented:
OK, good solution!

Thanks. :+1:

Commit:
[pep8] - review : replace _logger.log(logging.DEBUG by _logger.debug( / use , instead of % for interpolation

yvaucher commented:
restarting travis for odoo/odoo as it failed on loading eggs

pedrobaeza commented:
Hi, @yvaucher, how do you restart Travis build? I think I can't do it.

yvaucher commented:
@pedrobaeza You need to be logged on travis-ci.org and in each build you will find a restart button

yvaucher commented:
It's green, thanks
:+1:

pedrobaeza commented:
@yvaucher, I have found it, it's inside each job. I was looking in the upper level.

Thanks.

bwrsandman commented on account_financial_report/report/parser.py:
```diff
@@ -26,16 +26,10 @@
-from openerp.tools.safe_eval import safe_eval as eval
+from openerp.report import report_sxw
+from openerp.tools.translate import _
+from openerp.osv import osv
```
`osv` is imported to use `osv.except_osv` at https://github.com/acsone/account-financial-reporting/blob/7.0-pep8/account_financial_report/report/parser.py#L616

Please import `orm` and use `orm.except_orm`

bwrsandman commented on account_financial_report/report/parser.py:
```diff
@@ -164,20 +164,26 @@ def special_period(self, periods):
+                and type(form['currency_id']) in (list, tuple) \
+                and form['currency_id'][0] or form['currency_id']
+        return self.pool.get('res.currency').browse(self.cr, self.uid,
+                                                    self.to_currency_id).name
```
Missing context propagation: it can be gotten with `self.pool['res.users'].context_get(self.cr, self.uid)`

bwrsandman commented on account_financial_report/report/parser.py:
```diff
@@ -292,29 +304,50 @@ def _get_analytic_ledger(self, account, ctx={}):
-            #~ FISCAL, SIN LOS PERIODOS ESPECIALES
+            # ~ TODO: CUANDO EL PERIODO ESTE VACIO LLENARLO CON LOS PERIODOS
+            # DEL EJERCICIO
+            # ~ FISCAL, SIN LOS PERIODOS ESPECIALES
```
Can this be translated?

bwrsandman commented on account_financial_report/report/parser.py:
```diff
@@ -374,7 +409,8 @@ def _get_journal_ledger(self, account, ctx={}):
                 })
-                print 'ACCOUNT NAME', am_obj.browse(self.cr, self.uid, det['am_id']).name
+                print 'ACCOUNT NAME', am_obj.browse(self.cr, self.uid,
+                                                    det['am_id']).name
```
Better to use logger

bwrsandman commented on account_financial_report/report/parser.py:
```diff
@@ -408,13 +447,15 @@ def _get_children_and_consol(cr, uid, ids, level, context={}, change_sign=False)
-            #~ ctx_end['periods'] = period_obj.search(self.cr, self.uid, [('fiscalyear_id','=',fiscalyear.id),('special','=',False)])
+            # ~ ctx_end['periods'] = period_obj.\
+            # search(self.cr, self.uid, [('fiscalyear_id','=',fiscalyear.id),
+            #                            ('special','=',False)])
```
Better off removing commented out code

bwrsandman commented on account_financial_report/wizard/wizard.py:
```diff
@@ -209,9 +270,12 @@ def _check_date(self, cr, uid, data, context=None):
-                    res[0]['date_start'], res[0]['date_stop']))
+            if (data['form']['date_to'] > res[0]['date_stop']
+                    or data['form']['date_from'] < res[0]['date_start']):
+                raise osv.except_osv(_('UserError'),
```
Pease use orm.except_orm

bwrsandman commented on account_financial_report/wizard/wizard.py:
```diff
@@ -209,9 +270,12 @@ def _check_date(self, cr, uid, data, context=None):
+            if (data['form']['date_to'] > res[0]['date_stop']
+                    or data['form']['date_from'] < res[0]['date_start']):
+                raise osv.except_osv(_('UserError'),
+                                     'Las fechas deben estar entre %s y %s'
```
Can this be translated?

bwrsandman commented on account_financial_report_horizontal/report/__init__.py:
```diff
@@ -1,3 +1,3 @@
-#import account_balance
-import account_balance_sheet
-import account_profit_loss
+# import account_balance
```
Remove this line

bwrsandman commented on account_financial_report_horizontal/report/common_report_header.py:
```diff
@@ -44,25 +43,25 @@ def _get_end_date(self, data):
-                'form']['period_from'][0]).name
+            return self.pool['account.period'].browse(
+                self.cr, self.uid,
+                data['form']['period_from'][0]).name
```
Missing context

bwrsandman commented on account_financial_report_webkit/report/common_reports.py:
```diff
@@ -348,67 +388,81 @@ def _compute_init_balance(self, account_id=None, period_ids=None, mode='computed
+        opening_period_selected = self.get_included_opening_period(
+            start_period)
         if not opening_period_selected:
             raise osv.except_osv(
```
Please use `orm.except_orm`

bwrsandman commented on account_financial_report_webkit/report/open_invoices.py:
```diff
@@ -120,7 +129,8 @@ def set_context(self, objects, data, ids, report_type=None):
+            new_ids, exclude_type=['view'], only_type=filter_type)
 
         if not account_ids:
             raise osv.except_osv(_('Error'), _('No accounts to print.'))
```
`orm.except_orm`

bwrsandman commented on account_financial_report_webkit_xls/report/partner_ledger_xls.py:
```diff
@@ -21,15 +21,14 @@
-#import logging
-#_logger = logging.getLogger(__name__)
+# import logging
+# _logger = logging.getLogger(__name__)
```
Please just remove thes two commented lines

bwrsandman commented on account_financial_report_webkit_xls/report/trial_balance_xls.py:
```diff
@@ -21,49 +21,56 @@
-#import logging
-#_logger = logging.getLogger(__name__)
+# import logging
+# _logger = logging.getLogger(__name__)
```
Please just remove these two lines

bwrsandman commented on account_financial_report_webkit_xls/wizard/general_ledger_wizard.py:
```diff
@@ -21,12 +21,13 @@
-#import logging
-#_logger = logging.getLogger(__name__)
+# import logging
+# _logger = logging.getLogger(__name__)
```
Please remove these two lines

bwrsandman commented on account_financial_report_webkit_xls/wizard/open_invoices_wizard.py:
```diff
@@ -21,12 +21,13 @@
-#import logging
-#_logger = logging.getLogger(__name__)
+# import logging
+# _logger = logging.getLogger(__name__)
```
Please remove these two lines

bwrsandman commented on account_financial_report_webkit_xls/wizard/partners_balance_wizard.py:
```diff
@@ -21,12 +21,13 @@
-#import logging
-#_logger = logging.getLogger(__name__)
+# import logging
+# _logger = logging.getLogger(__name__)
```
Please remove these two lines

bwrsandman commented on account_financial_report_webkit_xls/wizard/partners_ledger_wizard.py:
```diff
@@ -21,12 +21,13 @@
-#import logging
-#_logger = logging.getLogger(__name__)
+# import logging
+# _logger = logging.getLogger(__name__)
```
Please remove these two lines

bwrsandman commented on account_financial_report_webkit_xls/wizard/trial_balance_wizard.py:
```diff
@@ -21,12 +21,13 @@
-#import logging
-#_logger = logging.getLogger(__name__)
+# import logging
+# _logger = logging.getLogger(__name__)
```
Please remove these two lines

bwrsandman commented on account_journal_report_xls/account_journal.py:
```diff
@@ -59,17 +57,18 @@ def _report_xls_fields(self, cr, uid, context=None):
+            # 'journal',           # account.journal,name
+            # 'journal_code',      # account.journal,code
+            # 'analytic_account',       # account.analytic.account,name
+            # 'analytic_account_code',  # account.analytic.account,code
```
Please remove these commented out lines

bwrsandman commented:
Thank you @laetitia-gangloff for this not-so-small task. I have pointed out a few lines which may have been overlooked.

Honestly, this repo may need an audit, I saw a lot of problems:
- Long, indented functions
- Missing context propagations
- Non-english comments
- Non-english/non-translatable strings to the end-user
- Non-english variable names
- Coverage needs to be increased
- Duplicated code which can be factored
- A lot of commented out code
- I am sure there's a lot more

yvaucher commented:
I think this should be done in another PR.

Let's don't mix PEP8 greenifying and refactoring I would suggest to accept this PR

lepistone commented:
:+1: I also approve this as a lint-only pull request. Thanks @laetitia-gangloff!

We can open a new issue pointing to @bwrsandman's remarks and a related pull request.

nbessi commented:
:+1: agreed with @yvaucher and @lepistone

bwrsandman commented:
I created an issue (#10) for auditing the modules.

pedrobaeza commented:
I proceed then with this merge and let the audit for another PR.

Regards.
